### DISCUSSION TITLE
Default value using useQueryStates

### DISCUSSION THREAD

**Discussion by Emiliano-Bucci** [47ng/nuqs#823] [General] [answered]
Hi! I'm using useQueryStates to simplify logic of usage of many and multiple filters, and most of them are list of ids, i.e.

towns=1,2,3,4,5

The problem is that when I call **setFilters({ towns: [] })** even if I've setted **clearOnDefault=true** the query param is not removed from url, remaining ?towns=

Is there a way to prevent this?

Thanks!

**Comment by franky47** [comment, marked as answer]
Can you show me your hook declaration please?

You could be missing a .withDefault([]) on the parseAsArrayOf.

**Reply by Emiliano-Bucci** [reply]
Hi, and thanks for the quick answer!

I was actually missing the `.withDefault([])`, sorry and thanks! :)
